src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -4096,6 +4099,23 @@ public IndexingStatsAggregator GetLatestIndexingStat()
 
         public abstract void SaveLastState();
 
+        
+        protected void HandleSourceDocumentIncludedInMapOutput()
+        {
+            if (_alreadyNotifiedAboutIncludingDocumentInOutput || SourceDocumentIncludedInOutput == false || PerformanceHintsConfig.AlertWhenSourceDocumentIncludedInOutput == false)
+                return;
+            
+            DocumentDatabase.NotificationCenter.Add(PerformanceHint.Create(
+                DocumentDatabase.Name,
+                $"Index '{Name}' is including the origin document in output.",
+                $"Please verify index definitions and consider a re-design of your entities or indexes for better indexing performance.",
```

Comment:
I wonder if we can express it better. What do you think about something like:

`Putting the whole document as one of the fields of the index entry isn't usually intentional. Please verify your index definition for better indexing performance`

?

Reply:
Yeah, this is better.